installer/src/App.jsx

```diff
@@ -1,147 +1,395 @@
+    const handleError = (error) => {
+        dispatch({ type: 'GENERIC_ERROR', error });
+    };
 
-      let boards = boardList
-        .filter(row => row['matching_boards']?.find(o => fqbn_list.includes(o['fqbn'])))
-        .map(board => ({
-          name: board['matching_boards'][0]["name"],
-          fqbn: board['matching_boards'][0]["fqbn"],
-          port: board['port']["address"],
-        }))
+    useEffect(() => {
+        textAreaRef.current.scrollTop = textAreaRef.current.scrollHeight;
+    }, [state.stdout]);
 
-      setBoards(boards)
-    }
+    useEffect(() => {
+        const getVersion = async () => {
+            const command = Command.sidecar(
+                '../resources/arduino-cli/arduino-cli',
+                ['version', '--format', 'json']
+            );
+            const output = await command.execute();
+            const versionObject = JSON.parse(output.stdout);
+            setVersion(
+                `${versionObject.VersionString} - ${versionObject.Date} `
+            );
+        };
+        getVersion().catch(handleError);
+    }, []);
 
-    getBoards(['arduino:mbed_nano:nano33ble']).catch(console.error)
-  })
+    /* Checking if there are connected boards every 3 seconds */
+    useEffect(() => {
+        const interval = setInterval(async () => {
+            try {
+                const connectedBoards = await getBoards([
+                    'arduino:mbed_nano:nano33ble',
+                ]);
+                if (connectedBoards.length > 0) {
+                    dispatch({
+                        type: 'BOARDS_RETRIEVED',
+                        boards: connectedBoards,
+                    });
+                } else {
+                    dispatch({ type: 'BOARDS_DISCONNECTED' });
+                }
+            } catch (error) {
+                console.error(error);
+            }
+        }, 3000);
 
-  const upload = async () => {
-    setUploading(true)
+        return () => clearInterval(interval);
+    }, []);
 
-    let board = boards.find(x => x.name === selectedBoard)
+    const updateLeftServo = (ev) => {
+        setLeftServo(ev.currentTarget.value);
+    };
+    const updateRightServo = (ev) => {
+        setRightServo(ev.currentTarget.value);
+    };
 
+    const updateName = (ev) => {
+        setName(ev.currentTarget.value);
+    };
 
-    let dstPath = await invoke("append_config", { binPath: "../resources/sketch.bin", cfg: "test" }).catch(console.error);
+    const asyncUpload = async () => {
+        dispatch({ type: 'UPLOAD_STARTED' });
+        let board;
+        let dstPath;
+        try {
+            board = state.boards.find((x) => x.name === state.selectedBoard);
+            dstPath = await invoke('append_config', {
+                binPath: '../resources/sketch.bin',
+                ble_name: name,
+                cfg: name, // legacy, remove when msgpack stuff is completed
+                left_servo: leftServo,
+                right_servo: rightServo,
```

Comment:
this will be a single `config` parameter with a config struct, see here https://github.com/arduino/include-robot/pull/57/files#diff-dcdb6a8b305a3e53d6544ed0bad91742d5aacc22644f29737db85b2085221265R69

Reply:
Good catch!